### PR TITLE
Add Support for Fetch First Based Paging

### PR DESCRIPTION
**Pull request by jeffgbutler**

### CODE REVIEW

File: src/main/java/org/mybatis/dynamic/sql/select/render/FetchFirstPagingModelRenderer.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+    
+    private Optional<FragmentAndParameters> renderOffsetOnly(Long offset) {
+        return FragmentAndParameters.withFragment("offset " + renderPlaceholder(OFFSET_PARAMETER)) //$NON-NLS-1$
+                .withParameter(OFFSET_PARAMETER, offset)
+                .buildOptional();
+    }
+    
+    private Optional<FragmentAndParameters> renderOffsetAndFetchFirstRows(Long offset, Long fetchFirstRows) {
+        return FragmentAndParameters.withFragment("offset " + renderPlaceholder(OFFSET_PARAMETER) //$NON-NLS-1$
```
Comment (kazuki43zoo) [inline src/main/java/org/mybatis/dynamic/sql/select/render/FetchFirstPagingModelRenderer.java]: Hi @jeffgbutler ,

The "offset #{_offset}" does not work on Oracle Database(I've tried on 19.3.0).

```
SQL*Plus: Release 19.0.0.0.0 - Production on Wed Jun 26 15:33:39 2019
Version 19.3.0.0.0

Copyright (c) 1982, 2019, Oracle.  All rights reserved.

Last Successful login time: Wed Jun 26 2019 15:33:05 +00:00

Connected to:
Oracle Database 19c Enterprise Edition Release 19.0.0.0.0 - Production
Version 19.3.0.0.0

SQL> 
SQL> 
SQL> select * from dual offset 0 fetch first 10 rows only;
select * from dual offset 0 fetch first 10 rows only
                          *
ERROR at line 1:
ORA-00933: SQL command not properly ended
```

Above error can fix by change to **"offset #{_offset} rows"** as follow:

```
SQL> 
SQL> select * from dual offset 0 rows fetch first 10 rows only;

D
-
X
```

Probably, the Oracle cannot omit "rows / row" phrase.

### Related links

* PostgreSQL spec is https://www.postgresql.org/docs/11/sql-select.html.

* H2 spec is https://www.h2database.com/html/commands.html#select.

* HSQLDB spec is http://hsqldb.org/doc/2.0/guide/dataaccess-chapt.html#dac_slicing.

* DB2 spec is https://www.ibm.com/support/knowledgecenter/en/SSEPEK_12.0.0/sqlref/src/tpc/db2z_sql_offsetclause.html.

* SQLServer spec is https://docs.microsoft.com/ja-jp/sql/t-sql/queries/select-order-by-clause-transact-sql?view=sql-server-2017.

* Mimer SQL spec is https://download.mimer.com/pub/developer/docs/html_110/Mimer_SQL_Engine_DocSet/select_spec.html#1119803.

Comment (jeffgbutler) [reply src/main/java/org/mybatis/dynamic/sql/select/render/FetchFirstPagingModelRenderer.java]: @kazuki43zoo  Interesting - and thanks for the review.

It looks like offset #{...} rows" should work everywhere - am I right about that?

Comment (jeffgbutler) [reply src/main/java/org/mybatis/dynamic/sql/select/render/FetchFirstPagingModelRenderer.java]: I see you have done it in the SQL builder class. I will do the same. Thanks again!

Comment (kazuki43zoo) [reply src/main/java/org/mybatis/dynamic/sql/select/render/FetchFirstPagingModelRenderer.java]: 👍

**Pull request merged**